### DISCUSSION TITLE
하나은행 - mbp.hanabank.com

### DISCUSSION THREAD

**Discussion by piecemaker247** [List-KR/community#4] [General]
Hi, I am Adgaurd users in korea and i want to request something.
"mbp.hanabank.com" is one of my banking application domain but adguard keep blocking that domain.

I can't use my banking app now... please exclusde "mbp.hanabank.com"

---

### [Go to @ameshkov's answer.](https://github.com/List-KR/List-KR/discussions/732#discussioncomment-6292291)

**Comment by piquark6046** [comment]
Can I see your filtering request when the popup appears?

**Comment by piecemaker247** [comment]
Adguard v4.0 for android.

And this is screen shot when blocked

2023년 6월 22일 (목) 오후 5:36, PiQuark6046 ***@***.***>님이 작성:

**Reply by piquark6046** [reply]
Replying via email does not attach your attachment.
Please login into GitHub and reply at GitHub.

**Reply by piecemaker247** [reply]
![Screenshot_20230622-151223](https://github.com/List-KR/List-KR/assets/48271337/4d253b8a-0758-4de6-9299-68c5af8a94fc)
This is screen shot

**Reply by piquark6046** [reply]
I meant the filtering log which is located in AdGuard for Android -> `통계` tab -> `최근 활동`.

**Reply by piecemaker247** [reply]
![Screenshot_20230622-181412_AdGuard](https://github.com/List-KR/List-KR/assets/48271337/6035c80b-d9af-473f-be15-2177bd6141d5)
![Screenshot_20230622-181339_AdGuard](https://github.com/List-KR/List-KR/assets/48271337/f18aa8a4-867c-4e04-8497-acb508f9e654)

**Reply by piecemaker247** [reply]
![Screenshot_20230622-181049_AdGuard](https://github.com/List-KR/List-KR/assets/48271337/df1f077e-f831-4336-8ff7-7439a05fdb9d)
![Screenshot_20230622-181038_AdGuard](https://github.com/List-KR/List-KR/assets/48271337/9fa171f3-4400-435a-b8a9-8c9873f3d6f9)

**Reply by piquark6046** [reply]
Please use Cloudflare DNS server instead of AdGuard DNS server.

**Reply by piecemaker247** [reply]
Why does it happen? is it becuase that domain(mbp.hanabank.com) is not registered in AdGaurd DNS server?

**Reply by piquark6046** [reply]
I left comment related to this issue at AdGuard Slack.

**Reply by piecemaker247** [reply]
Thanks. Can i get any follow-up about this issue? I think a lot of users in Korea will suffer the same problem.
It seems that domain should be registered in your DNS server.

**Reply by ameshkov** [reply]
Could you please open `https://dns.adguard-dns.com/info.txt`? We need to see which server you're routed to.

**Reply by piecemaker247** [reply]
dns2-dp-tok-4

**Reply by ameshkov** [reply]
Well, I am able to resolve that domain. The only idea I have is that resolution takes too much time, longer than the default timeout that we're using.

Could you please try increasing it? Head to General -> Advanced -> Low-level settings and increase the DNS request timeout.

**Reply by piecemaker247** [reply]
I doubleed the timeout but still it happens.(5000 -> 10000mili seconds)
Sever was dns2-dp-tok-1

**Reply by piecemaker247** [reply]
![Screenshot_20230623-094123](https://github.com/List-KR/List-KR/assets/48271337/0a94af26-7ab3-466f-a2a3-758f3527cde6)
![Screenshot_20230623-094134_AdGuard](https://github.com/List-KR/List-KR/assets/48271337/0a93aa7d-14db-471a-9472-99932f7a834e)

It seems it happens when your DNS server respond with SERVFAIL. Why does it give SERVFAIL??
Dns server was  dns2-dp-tok-4

**Reply by ameshkov** [reply]
Most likely servfail is a consequence of a timeout error. And the timeout error is the consequence of `mbp.hanabank.com` blocking our name server. It will take quite some time to resolve properly, we now switched it to a fallback on our side.

**Reply by piecemaker247** [reply]
Thanks for reply. One more thing i want to ask. It seems sometimes the connection is ok, and sometimes it blocks. Is it because your server setting is different from servers?? Like some servers don't block "mbp.hanabank.com" but others do. Please let me know when you settle the issue.

**Reply by ameshkov** [reply]
It should work right now.

> It seems sometimes the connection is ok, and sometimes it blocks

Sometimes our servers are serving response from the cache, sometimes the cached record expired and they need more time to try to resolve that domain, that's when you can run into the timeout issue.

**Reply by piecemaker247** [reply]
It seems working now!! Thank you so much for your cooperation! I will contact again if the problem happens again!

**Reply by piquark6046** [reply]
@ameshkov 
https://github.com/AdguardTeam/AdguardFilters/issues/154370#issuecomment-1605808438

**Comment by piecemaker247** [comment]
Blocking "mbp.hanabank.com" happens again.... Have you changed your server setting??? I can't use my application.
"mbp.hanabank,com" DNS response is "SERVFAIL"

**Reply by piquark6046** [reply]
For now, please use Cloudflare DNS server instead of AdGuard DNS.

**Reply by ameshkov** [reply]
It's rather hard to maintain the fallback all the time. It'd help if the root cause was resolved.

**Reply by ameshkov** [reply]
Added some details here: https://github.com/List-KR/List-KR/discussions/732#discussioncomment-6292291

**Comment by kimjs0911** [comment]
> Hi, I am Adgaurd users in korea and i want to request something. "mbp.hanabank.com" is one of my banking application domain but adguard keep blocking that domain.
> 
> I can't use my banking app now... please exclusde "mbp.hanabank.com"

i agree also. all bank app can not be used

**Reply by piquark6046** [reply]
Does Selecting AdGuard DNS cause your issue?
If so, can selecting Cloudflare DNS in AdGuard application resolve your issue?

**Reply by piquark6046** [reply]
@kimjs0911 Can you tell me what the bank apps that being used?

**Reply by tosmile78** [reply]
Probably, this one: https://play.google.com/store/apps/details?id=com.kebhana.hanapush&hl=en_US

It bothers millions of users. fix this ASAP, please.

**Comment by bang1myoung** [comment]
If I register private dns as dns.adgurad.com on my Android phone, I can't access mbp.hanabank.com . Please take care of it quickly.

**Comment by tosmile78** [comment]
I understand Adgurad protects people agains many harm sites.
Though, it sometimes blocks harmless sites, like 'mbp.hanabnak.com'.
It is the domain of the major bank in South Korea.
I hope you fix this problem ASAP. 
Thanks for your effort in advance.

**Comment by piquark6046** [comment]
@bang1myoung @tosmile78

Actually, I have not the permission related to manage AdGuard DNS server.
I heard that an employee of hanabnak's zendesk ticket related to the issue was submitted and is being processed.
@ameshkov will handle the issue.

**Reply by bang1myoung** [reply]
May I know when it will be done?

**Reply by piquark6046** [reply]
OK. If I hear that, I will leave a message.

**Comment by ameshkov** [comment]
Unfortunately, the situation is reversed. This is not us blocking `mbp.hanabank.com`, but their nameservers are actively preventing AdGuard DNS from resolving this domain.

Technically, they block all connections from our Japan DNS recursor server (`156.146.34.244`) to their nameservers (`ns13.kebhana.com`, `ns12.kebhana.com`, `ns11.kebhana.com`).

We have no idea why they're doing this and who to contact to get it resolved. If you can let them know about the issue details through the support service, it would be greatly appreciated. We're maintaining a temporary solution, but it is not a stable solution and eventually we need this issue to get resolved.

@piquark6046 could you please mark this response as a correct answer so that other people could see it.

**Reply by piquark6046** [reply]
Because this category is not a Q&A, I cannot mark response as a correct answer.
But, I can pin this discussion and add its URL into top message of the discussion.

**Reply by bang1myoung** [reply]
Then tell me how to solve it

**Reply by bang1myoung** [reply]
Currently, mbp.hanabank.com is normally accessed. Is this not going to happen in the future?

**Reply by ameshkov** [reply]
There's a workaround solution in place, we'll keep it until Hanabank resolves the root cause on their side.

@piquark6046 pinning would be great.

**Reply by piquark6046** [reply]
Pinned 42 minutes ago.

**Reply by piecemaker247** [reply]
@ameshkov workaround solution means it will be temporary?? You can't keep it if there is no burden to hold?

**Reply by ameshkov** [reply]
We can keep it for quite some time, but we're interested in resolving this issue by normal means.

Also, I don't think that `mbp.hanabank.com` is the only affected domain. Most likely there are more domains that rely on the same nameservers, we're just not yet aware of them.
